cachi2/core/package_managers/bundler/main.py

```diff
@@ -24,5 +34,67 @@ def fetch_bundler_source(request: Request) -> RequestOutput:
 
 def _resolve_bundler_package(package_dir: RootedPath, output_dir: RootedPath) -> list[Component]:
     """Process a request for a single bundler package."""
-    parse_lockfile(package_dir)
+    dependencies = parse_lockfile(package_dir)
+    _get_main_package_name_and_version(package_dir, dependencies)
     return []
+
+
+def _get_main_package_name_and_version(
+    package_dir: RootedPath,
+    dependencies: ParseResult,
+) -> tuple[str, Optional[str]]:
+    """
+    Get main package name and version.
+
+    The main package is the package that is being processed by cachi2.
+    Not any of its dependencies.
+    """
+    path_dependencies = [dep for dep in dependencies if isinstance(dep, PathDependency)]
+    name_and_version = _get_name_and_version_from_path_deps(path_dependencies)
+    if name_and_version is not None:
+        return name_and_version
+
+    log.info("Failed to infer name and version from gemspec file")
+
+    # fallback to origin remote
+    try:
+        name = _get_repo_name_from_origin_remote(package_dir)
+    # if the git repository does not have an origin remote
+    except UnsupportedFeature:
+        raise PackageRejected(
+            reason="Failed to infer package name from origin remote",
+            solution=(
+                "Please specify package name and version in a way that Cachi2 understands,\n"
+                "or make sure that the directory Cachi2 is processing is a git repository with\n"
+                "an 'origin' remote, in which case Cachi2 will infer the package name from the remote URL."
+            ),
+        )
+
+    return name, None
+
+
+def _get_name_and_version_from_path_deps(
```

Comment:
This is quite an unfortunate name, because one really has to dive deeper to understand where should the name come from. It should be called more straightforward, something like `_get_name_and_version_from_gemspec` because it is essentially what it does and where the information is coming from. It's not a blocker though.

Reply:
Please note, that this function does not work with gemspec, its only argument is a list of path dependencies. Moreover, gemspec is not mentioned in the code at all, all the code does is some iteration over some path dependencies! A more self-documenting code with a gemspec mentioned in function name could look like this:
```python
def select_path_dep_for_gemspec(pds):
      return next(pd for pd in pds if pd.path == ".")
gemspec_dep = select_path_dep_for_gemspec(path_dependencies)
return gemspec_dep.name, gempec_dep.version if gemspec_dep is not None else None
```
If expressed like this the name could have mentioned gemspec, it would confuse future maintainers otherwise.

Reply:
> Please note, that this function does not work with gemspec

I know that it doesn't, but it's pretty much the first thing it mentions in the docstring, so there either is a relation or there isn't.

> ```
> def _get_name_and_version_from_gemspec(path_dependencies):
>     def select_path_dep_for_gemspec(pds):
>          return next(pd for pd in pds if pd.path == ".")
>     gemspec_dep = select_path_dep_for_gemspec(path_dependencies)
>     return gemspec_dep.name, gempec_dep.version if gemspec_dep is not None else None
> ```
No objection to ^this. With a tiny nitpick on working over all deps, not just path deps, so that we don't loop it "twice".

Reply:
Another option could be to slightly tweak the name: it is not 'from', it is 'for' a gemspec (which is now represented as a path dependency). Resulting name might raise some eyebrows, but reads as a succinct description of what is going on inside the function: `_get_name_and_version_for_gemspec_from(path_dependencies)`. In this case its definition matters much less.

Reply:
I renamed it to `get name_and_version_from_lockfile` for now. I think it is along the lines of the docstring and dependencies_2 as an argument. But I leave this unresolved.